netbox/extras/api/views.py

```diff
@@ -214,53 +214,36 @@ def render(self, request, pk):
 # Scripts
 #
 
-class ScriptViewSet(ViewSet):
+class ScriptViewSet(ModelViewSet):
     permission_classes = [IsAuthenticatedOrLoginNotRequired]
+    queryset = Script.objects.all().prefetch_related('jobs')
+    serializer_class = serializers.ScriptSerializer
+    # filterset_class = filtersets.ScriptFilterSet
+
     _ignore_model_permissions = True
     schema = None
     lookup_value_regex = '[^/]+'  # Allow dots
 
     def _get_script(self, pk):
-        try:
-            module_name, script_name = pk.split('.', maxsplit=1)
-        except ValueError:
-            raise Http404
-
-        module, script = get_module_and_script(module_name, script_name)
-        if script is None:
-            raise Http404
+        # check if includes '.' for old module.script lookup
+        if '.' in pk:
+            try:
+                module_name, script_name = pk.split('.', maxsplit=1)
+            except ValueError:
+                raise Http404
+
+            module, script = get_module_and_script(module_name, script_name)
+            if script is None:
+                raise Http404
+        else:
+            pk = int(pk)
```

Comment:
This can be removed; with the new URL redirects, the PK will always be an integer.